InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java

```diff
@@ -0,0 +1,98 @@
+package com.ccs.inventorymanagement.route;
+
+import com.ccs.inventorymanagement.config.RouteConfig;
+import com.ccs.inventorymanagement.domain.Clothing;
+import com.ccs.inventorymanagement.domain.Item;
+import com.ccs.inventorymanagement.service.ClothingService;
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.server.HandlerFunction;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+import java.util.function.Function;
+
+public class UpdateClothingHandler implements HandlerFunction<ServerResponse> {
+
+    private final ClothingService service;
+
+    public UpdateClothingHandler(ClothingService service) {
+        this.service = service;
+    }
+
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(Request.class)
+                .flatMap(new Function<Request, Mono<? extends Clothing>>() {
+                    @Override
+                    public Mono<? extends Clothing> apply(Request request) {
+                        return service.update(Clothing.builder()
+                                .id(UUID.fromString(serverRequest.pathVariable(RouteConfig.ID_VARIABLE)))
+                                .name(request.getName())
+                                .description(request.getDescription())
+                                .condition(request.getCondition())
+                                .status(Item.Status.PRESENT)
+                                .brand(request.getBrand())
+                                .color(request.getColor())
+                                .type(request.getType())
+                                .gender(request.getGender())
+                                .size(request.getSize())
+                                .build());
+                    }
+                }).flatMap(new Function<Clothing, Mono<? extends ServerResponse>>() {
+                    @Override
+                    public Mono<? extends ServerResponse> apply(Clothing clothing) {
+                        return ServerResponse.ok()
+                                .body(BodyInserters.fromValue(Response.from(clothing)));
+                    }
+                })
+                .onErrorResume(ex -> ServerResponse.badRequest().build());
```

Comment:
Theres generally two error scenarios that should be covered.
1. onErrorResume() will catch every error downstream from itself, most of these are going to be internal server errors.
2. Under onErrorResume() you could put a switchIfEmpty() operator to catch the scenario where the service cant find the entity with the provided ID. You would want to return a 404 not found in that case.

Reply:
Updated